clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1512,7 +1513,9 @@ public void unsubscribe() {
                 }
                 resetGroupMetadata();
             }
-            subscriptions.unsubscribe();
+        } catch (Exception e) {
+            log.error("Unsubscribe failed", e);
+            throw e;
```

Comment:
is this right? unsubscribe only throws two exceptions, i.e. the illegal state and illegal argument exception.

for InterruptedException, I wonder if we should call Thread.Interrupted() so when can cleanly shutdown the consumer. wdyt?

Also InterruptedException can be thrown everywhere now, i guess the reason we are seeing this is for the obvious reason, the worker calls unsubscribe before shutting down.